6_run_fit/test_run/fit_data.py

```diff
@@ -0,0 +1,601 @@
+    Must be the same as that used for constructing SMEE force field and topology
+    objects.
+--n-epochs : int, optional
+    Number of training epochs (default: 1000).
+--learning-rate : float, optional
+    Learning rate for Adam optimizer (default: 0.001).
+--batch-size : int, optional
+    Batch size for training (default: 500).
+
+Examples
+--------
+Train a force field with default parameters:
+$ python fit_data.py --data-dir ./data-train --filename-forcefield smee_force_field.pkl \\
+    --filename-topo-dict smee_topology_dict.pkl --offxml openff-2.2.1.offxml
+
+Input Dataset Structure
+-----------------------
+data-dir/ : dir
+├── dataset_info.json          # Dataset metadata and schema
+├── state.json                 # Dataset state information
+└── data-00000-of-00001.arrow  # Actual data in Apache Arrow format
+    ├── smiles (str)           # SMILES molecular representation
+    ├── coords (list)          # Flattened 3D coordinates
+    ├── energy (list)          # Flattened energy values
+    └── forces (list)          # Flattened force vectors
+
+Required .pkl Files
+-------------------
+filename-forcefield.pkl : file
+    Serialized SMEE TensorForceField object containing force field parameters
+    and potentials ready for optimization. Generated from OFFXML force field.
+
+filename-topo-dict.pkl : file
+    Dictionary mapping SMILES strings to SMEE TensorTopology objects.
+    Contains molecular topology information (bonds, angles, dihedrals) for
+    each unique molecule in the dataset.
+
+Output Structure
+----------------
+Creates the following outputs in current working directory:
+- my-smee-fit/                 # Training logs and checkpoints
+  ├── events.out.tfevents.*    # TensorBoard event files
+  ├── force-field-epoch-*.pt   # Checkpoints every 10 epochs
+  └── final-force-field.pt     # Final optimized force field
+- final-force-field.offxml     # Optimized force field in OFFXML format
+"""
+
+import pathlib
+import math
+from collections import defaultdict
+import pickle
+
+from tqdm import tqdm
+import torch
+import argparse
+import smee
+import datasets
+import descent
+import descent.train
+import descent.targets
+import descent.targets.energy
+from loguru import logger
+
+import tensorboardX
+import more_itertools
+
+from openff.toolkit import ForceField
+
+parameters = {
+    "Bonds": descent.train.ParameterConfig(
+        cols=["k", "length"],
+        scales={"k": 1e-2, "length": 1.0},  # normalize so roughly equal
+        limits={"k": [0.0, None], "length": [0.0, None]},
+        # the include/exclude types are Interchange PotentialKey.id's -- typically SMIRKS
+        # include=[], <-- bonds to train. Not specifying trains all
+        # exclude=[], <-- bonds NOT to train
+    ),
+    #    "Angles": descent.train.ParameterConfig(
+    #        cols=["k", "angle"],
+    #        scales={"k": 1e-2, "angle": 1.0},
+    #        limits={"k": [0.0, None], "angle": [0.0, math.pi]},
+    #    ),
+}
```

Comment:
Does this mean "Only train bonds" or "train bonds, angles, propers and impropers, but use the default settings for everything other than bonds"? I'm like 75% sure it's the former...